src/ensenso.cpp

```diff
@@ -52,7 +52,25 @@ Result<NxLibItem> imageNode(NxLibItem stereo, std::optional<NxLibItem> monocular
 	return estd::error("failed to get image node: unknown image type: " + std::to_string(int(type)));
 }
 
-Result<Ensenso> Ensenso::open(std::string serial, bool connect_monocular, LogFunction logger, NxLibInitToken token) {
+Result<std::shared_ptr<Ensenso>> Ensenso::openSharedCamera(std::string serial, bool connect_monocular, LogFunction logger, NxLibInitToken token) {
+	Result<OpenCameraReturn> open_camera = Ensenso::open(serial, connect_monocular, logger, token);
+	if (!open_camera) {
+		return open_camera.error();
+	}
+
+	return std::make_shared<Ensenso>(Ensenso(std::get<0>(*open_camera),  std::get<1>(*open_camera), std::get<2>(*open_camera), std::get<3>(*open_camera)));
+}
+
+Result<Ensenso> Ensenso::openCamera(std::string serial, bool connect_monocular, LogFunction logger, NxLibInitToken token) {
+	Result<OpenCameraReturn> open_camera = Ensenso::open(serial, connect_monocular, logger, token);
+	if (!open_camera) {
+		return open_camera.error();
+	}
+
+	return Ensenso(std::get<0>(*open_camera),  std::get<1>(*open_camera), std::get<2>(*open_camera), std::get<3>(*open_camera));
```

Comment:
You can use [`std::apply`](https://en.cppreference.com/w/cpp/utility/apply) to make this easier:

```suggestion
	return std::apply(Ensenso::Ensenso, std::move(*open_camera));
```

Reply:
I lie, but you can use `std::make_from_tuple`:

```suggestion
	return std::make:from_tuple<Ensenso>(std::move(*open_camera));
```

Reply:
Hmm, doesn't seem to be easy to do something similar for `std::make_shared`.

Reply:
Actually, you could do:
```
std::make_shared<Ensenso>(std::make_from_tuple<Ensenso>(std::move(*open_camera)));
```

This should work as long as the move constructor isn't deleted.

Reply:
Using ``make_from_tuple`` does not compile the code, because the constructor is private.

Reply:
Oh, right. You could fix that with a private `PrivateToken` struct, but nevermind then.

Reply:
Could you create an issue to implement the ``PrivateToken``?

Reply:
I don't think it's necessary. The `std::get` isn't that bad. So I'd just stick with this.